platforms/android/library/src/main/java/io/element/android/wysiwyg/EditorStyledTextView.kt

```diff
@@ -135,37 +137,64 @@ open class EditorStyledTextView : AppCompatTextView {
         )
     }
 
+    // Tracks the initial touch coordinates of a touch event, they will be used to determine if a click event happened
+    private var initialTouchCoordinates: Pair<Float, Float>? = null
+    // Tracks whether a click event may be happening between touch event phases
+    private var maybeClickEvent = false
+
     override fun onTouchEvent(event: MotionEvent?): Boolean {
-        return when (event?.action) {
+        when (event?.action) {
             MotionEvent.ACTION_DOWN -> {
-                // Allow ACTION_UP to be called
+                // A press was detected, this may be a click
+                maybeClickEvent = true
+                // Call default touch handler to detect the start of normal taps and gestures
                 super.onTouchEvent(event)
-                true
+                // Store initial touch coordinates
+                initialTouchCoordinates = event.x to event.y
+                // Return that we handled the event, it'll allow us to receive `ACTION_UP` later
+                return true
             }
-            MotionEvent.ACTION_UP -> {
-                // Find selection matching the pointer coordinates
-                val offset = getOffsetForPosition(event.x, event.y)
-                // Look for clickable spans in that position
-                val spans = (text as? Spanned)?.getSpans(offset, offset, Any::class.java) ?: return false
-                val linkSpans = spans.filterIsInstance<LinkSpan>()
-                val pillSpans = spans.filterIsInstance<PillSpan>()
-                val customMentionSpans = spans.filterIsInstance<CustomMentionSpan>()
-                // Notify the link has been clicked
-                for (span in linkSpans) {
-                    if (span.url == null) continue
-                    onLinkClickedListener?.invoke(span.url)
+            MotionEvent.ACTION_MOVE -> {
+                val distance = initialTouchCoordinates?.let { hypot(it.first - event.x, it.second - event.y) } ?: 0f
```

Comment:
We could skip a this calculation too by doing `if (!maybeClickEvent) return` here.

Reply:
I guess we could, but I think `ACTION_MOVE` can only happen after `ACTION_DOWN`, so `maybeClickEvent` should always be true at that point. I can add it, just in case.

Reply:
I thought `ACTION_MOVE` could be called many times during scrolling before `ACTION_UP` so thought of skipping any after `maybeClickEvent` gets reset [here](https://github.com/matrix-org/matrix-rich-text-editor/pull/860/files/78f8d42c9aedd1de5975306afee64016cc95ebab..931486ff3433dc40c423d4534d96dcf6cb29fcf9#diff-06656006549b36b19cc7b9afbd701cd4e71f2832c84971b16316f0641bbb486eR161). Maybe I missed something though?

Reply:
No, you're right, I completely forgot it could be called several times. In any case, using the `GestureDetector` makes this code no longer needed, so I'll remove it.